### PR TITLE
Fix app message title, clicking and other small improvements

### PR DESCRIPTION
**Pull request by pedropregueiro**

### CODE REVIEW

File: packages/common/src/reducers/messages.js [hunk 1]
```diff
@@ -231,6 +231,7 @@ const deriveMessageType = (message) => {
     case 1:
       return "member-joined";
     case 2:
+    case 3:
```
Comment (pedropregueiro) [inline packages/common/src/reducers/messages.js:234]: Forgot to add this to handle other app message types. Might change later to it's own type, if we decide to handle apps vs. pure webhooks differently

File: packages/desktop/src/components/channel-message.js [hunk 2]
```diff
@@ -325,6 +325,7 @@ const ChannelMessage = React.memo(function ChannelMessage_({
         <AppMessage
           isHovering={isHovering}
           message={message}
+          onClickInteractiveElement={onClickInteractiveElement}
```
Comment (pedropregueiro) [inline packages/desktop/src/components/channel-message.js:328]: this feels odd, but not sure how best to do it without moving the whole `onClickInteractiveElement` method out of the `ChannelMessage` component. suggestions welcome!

File: packages/desktop/src/components/channel-message.js [hunk 3]
```diff
@@ -717,6 +718,42 @@ const MemberDisplayName = React.forwardRef(
...
+          color: color ?? theme.colors.pink,
+          fontWeight: "500",
+          display: "flex",
+          alignItems: "center",
+        })
+      }
+      {...props}
+    >
+      {displayName}
+      <span
```
Comment (pedropregueiro) [inline packages/desktop/src/components/channel-message.js:737]: stole the little App badge from slack:
Screenshot 2022-06-04 at 15 43 08

File: packages/common/src/reducers/messages.js [hunk 4]
```diff
@@ -263,12 +262,18 @@ export const selectMessage = createSelector(
     return selectMessage(state, message.reply_to);
   },
   (state) => state.user,
-  (message, author, repliedMessage, loggedInUser) => {
+  (state, messageId) => {
+    const message = state.messages.entriesById[messageId];
+    if (message == null || !message.app) return null;
+    return selectApp(state, message.app);
+  },
+  (message, author, repliedMessage, loggedInUser, app) => {
```
Comment (pedropregueiro) [inline packages/common/src/reducers/messages.js:270]: I said boring Sunday, but Mondays can also be a bore 😂 

This was helpful to improve this piece:
https://redux.js.org/usage/deriving-data-selectors

File: packages/desktop/src/components/channel-message.js (hunk 2 above)
Comment (werme) [reply packages/desktop/src/components/channel-message.js:332]: Would be prettier to compose the RichText one level up perhaps, but I think this is fine for now.

```js
// e.g.
<AppMessage
  message={
    <RichText
      blocks={message.content}
      onClickInteractiveElement={onClickInteractiveElement}
    />
  }
/>

**Review by werme — APPROVED** [review]
Sorry forgot about this! LGTM!

**Pull request merged**